### PR TITLE
nolibssp vim rebuild

### PR DESCRIPTION
**Pull request by satmandu**
- rebuild w/o libssp
- needs perl rebuild too. :/
- I'm going to stop updating `xxd_standalone`, since there are no changes to it...

Works properly:
- [x] x86_64
- [x] armv7l
- [x] i686

### CODE REVIEW

**Comment by saltedcoffii** [comment]
Why not use the libssp at #5629?

**Comment by satmandu** [comment]
@saltedcoffii The idea is to reduce core dependencies...

**Comment by saltedcoffii** [comment]
This package is really tiny and builds almost instantaneously. I'm guessing there's a good reason that vim and ruby link against libssp, the library provides signal processing stuff.

**Comment by satmandu** [comment]
https://wiki.osdev.org/Stack_Smashing_Protector

https://reviews.freebsd.org/D15687

**Comment by saltedcoffii** [comment]
"It is worth considering enabling it by default if your operating system is security conscious and you provide support."

This is now a question of whether we care about security. I would err on the side of security, as ChromeOS definitely values security. Thanks for those links by the way, I couldn't find any documentation on it.

**Comment by uberhacker** [comment]
We have needed to relax security a bit with `xdg_base` and `sed` to get around the tyrannical and heavy-handed ChromeOS security measures.

**Comment by saltedcoffii** [comment]
Yes, their security measures are quite harsh. Since libssp does no harm and is such a tiny package, I think we should use it.

**Comment by satmandu** [comment]
> Yes, their security measures are quite harsh. Since libssp does no harm and is such a tiny package, I think we should use it.

My initial skepticism was due to this pulling in all of GCC as a runtime dependency. But splitting off libssp into its own package which could be added to core does seem to make a lot of sense.

(LLVM depending upon GCC was/is especially annoying.)

**Comment by satmandu** [comment]
`vim` & `vim_runtime` here no longer need `perl` installed to work. `gvim` is built against the updated `perl` 5.32.1 in https://github.com/skycocker/chromebrew/pull/5665

**Comment by satmandu** [comment]
Merging this before https://github.com/skycocker/chromebrew/pull/5665 will break `gvim`, but then merging https://github.com/skycocker/chromebrew/pull/5665 before updating all perl packages will... you know.

**Comment by satmandu** [comment]
This now works with the current version of perl, and the new version, so this should be good to go.

File: packages/xxd_standalone.rb [hunk 1]
```diff
@@ -3,24 +3,24 @@
 class Xxd_standalone < Package
   description 'Hexdump utility from vim'
   homepage 'http://www.vim.org'
-  @_ver = '8.2.2725'
+  @_ver = '8.2.2771'
```
Comment (uberhacker) [inline packages/xxd_standalone.rb:6]: Can this be updated to 8.2.2783?

Comment (satmandu) [reply packages/xxd_standalone.rb:6]: Yes, but the source file hasn't changed. It compiled to literally the same binary.

**Pull request merged**